packages/EasyErrorHandler/src/Processors/ErrorCodesGroupProcessor.php

```diff
@@ -7,21 +7,24 @@
 use EonX\EasyErrorHandler\DataTransferObjects\ErrorCodeCategoryDto;
 use EonX\EasyErrorHandler\DataTransferObjects\ErrorCodesDto;
 use EonX\EasyErrorHandler\Interfaces\ErrorCodesGroupProcessorInterface;
-use EonX\EasyErrorHandler\Interfaces\ErrorCodesProviderInterface;
+use EonX\EasyErrorHandler\Interfaces\ErrorCodesProviderLocatorInterface;
 
 final class ErrorCodesGroupProcessor implements ErrorCodesGroupProcessorInterface
 {
     public function __construct(
-        private int $categorySize,
-        private ErrorCodesProviderInterface $errorCodesProvider
+        private readonly int $categorySize,
+        private readonly string $errorCodesSource,
```

Comment:
At the first glance we make the configuration too complex. For example you can set source to enum and leave auto discovery = false. Or source = enum and set error codes interface. Both these options won't work and make no sense actually. The easiest way to me is to support both source from the start. We can introduce a set of providers and go through them in the processor. Each provider uses its own logic to fetch the codes: interface provider uses `error_codes_interface` configuration (and returns an empty result if the configuration is not set) and the enum provider uses all `AsErrorCodes` implementation without any explicit configuration. Wdyt?

Reply:
You can't set the source manually it's set automatically in Configuration.php if errorCodeEnumsAutodiscovery = true.

Reply:
But I understood complexity. I'll remove errorCodeEnumAutodiscovery and use it only if errorCodesInterface is not set.

Reply:
Configuration changed. `errorCodeEnumAutodiscovery` removed. Now it's used only if errorCodesByInterfaceProvider returned empty array